discovery-provider/src/api/v1/playlists.py

```diff
@@ -345,6 +349,74 @@ def get(self):
         current_user_id, time = args.get("user_id"), args.get("time", "week")
         time = "week" if time not in ["week", "month", "year"] else time
 
+        strategy = trending_selector.get_strategy(TrendingType.PLAYLISTS)
+
+        # If we have a user_id, we call into `get_trending_playlist`
+        # which fetches the cached unpopulated tracks and then
+        # populates metadata. Otherwise, just
+        # retrieve the last cached value.
+        #
+        # If current_user_id,
+        # apply limit + offset inside the cached calculation.
+        # Otherwise, apply it here.
+        if current_user_id:
+            args = {
+                'time': time,
+                'with_tracks': True,
+                'limit': limit,
+                'offset': offset
+            }
+            decoded = decode_string_id(current_user_id)
+            args["current_user_id"] = decoded
+            playlists = get_trending_playlists(args, strategy)
+        else:
+            args = {
+                'time': time,
+                'with_tracks': True,
+            }
+            key = self.get_cache_key()
+            playlists = use_redis_cache(key, TRENDING_TTL_SEC, lambda: get_trending_playlists(args, strategy))
+            playlists = playlists[offset: limit + offset]
+
+        return success_response(playlists)
+
+@full_ns.route("/trending/ePWJD")
+class FullTrendingPlaylistsSecondary(Resource):
+    @full_ns.expect(full_trending_parser)
```

Comment:
I think this could be a little more consolidated. the logic in the "secondary"/"alternatives" is almost the same as the default. we can probably abstract this out of the api/v1/playlists.py into a helper